src/python/pants/backend/python/macros/poetry_requirements.py

```diff
@@ -449,6 +450,8 @@ async def generate_from_python_requirement(
             target_name=request.template_address.target_name,
             relative_file_path=pyproject_rel_path,
         ),
+        name_explicitly_set=request.generator_name_explicitly_set,
```

Comment:
These `TargetGeneratorSourcesHelperTarget` instances should be able to use the default afaict, because they're "private", and the code a few lines up always sets a name. The generator target itself is the only one which should potentially see the warning.

Reply:
My thinking was we need to know when references to generated targets will be invalid, e.g. `3rdparty#Django` becomes `3rdparty:reqs#Django`. It's convenient to store the info directly on the generated target. But then I realized that's not enough info...we need to know if the target generator violates one of our safe cases.

So I'll revert.